src/test/java/core/basesyntax/service/impl/LineParserImplTest.java

```diff
@@ -0,0 +1,95 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.service.LineParser;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class LineParserImplTest {
+
+    private static final List<FruitTransaction> EXPECTED_TRANSACTIONS = List.of(
+            new FruitTransaction(FruitTransaction.Operation.BALANCE, "banana", 20),
+            new FruitTransaction(FruitTransaction.Operation.BALANCE, "apple", 100),
+            new FruitTransaction(FruitTransaction.Operation.SUPPLY, "banana", 100),
+            new FruitTransaction(FruitTransaction.Operation.PURCHASE, "banana", 13),
+            new FruitTransaction(FruitTransaction.Operation.RETURN, "apple", 10),
+            new FruitTransaction(FruitTransaction.Operation.PURCHASE, "apple", 20),
+            new FruitTransaction(FruitTransaction.Operation.PURCHASE, "banana", 5),
+            new FruitTransaction(FruitTransaction.Operation.SUPPLY, "banana", 50));
+    private LineParser lineParser;
+
+    @BeforeEach
+    void setUp() {
+        lineParser = new LineParserImpl();
+    }
+
+    @Test
+    void createListOfTransactions_validLines_ok() {
+        List<String> validLines = List.of("type.fruit.quantity",
+                "b.banana.20",
+                "b.apple.100",
+                "s.banana.100",
+                "p.banana.13",
+                "r.apple.10",
+                "p.apple.20",
+                "p.banana.5",
+                "s.banana.50");
+        List<FruitTransaction> actualTransactions = lineParser
+                .createListOfTransactions(validLines);
+        assertEquals(EXPECTED_TRANSACTIONS, actualTransactions);
+    }
+
+    @Test
+    void createListOfTransactions_invalidLines_notOk() {
+        List<String> invalidLines = List.of("type.fruit.quantity",
+                "b.banana.20",
+                "b.apple.100",
+                "s.banana.r",
+                "p.banana.13",
+                "r.apple.10",
+                "p.apple.-20",
+                "p.banana.5",
+                "s.banana.50",
+                "x.orange.30");
```

Comment:
You duplicate strings, you can create template and reuse it in every test what you need and make some modification for templates, it makes code more clean